tracer/src/Datadog.Trace/Ci/CiEnvironment/GitCommandGitInfoProvider.cs

```diff
@@ -0,0 +1,144 @@
+// <copyright file="GitCommandGitInfoProvider.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.IO;
+using System.Linq;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.Ci.CiEnvironment;
+
+internal sealed class GitCommandGitInfoProvider : GitInfoProvider
+{
+    private GitCommandGitInfoProvider()
+    {
+    }
+
+    public static IGitInfoProvider Instance { get; } = new GitCommandGitInfoProvider();
+
+    private static DateTimeOffset UnixTimeStampToDateTime(long unixTimeStamp)
+    {
+        const long unixEpochTicks = TimeSpan.TicksPerDay * 719162; // 621,355,968,000,000,000
+        return new DateTimeOffset((unixTimeStamp * TimeSpan.TicksPerSecond) + unixEpochTicks, TimeSpan.Zero);
+    }
+
+    protected override bool TryGetFrom(DirectoryInfo gitDirectory, out IGitInfo? gitInfo)
+    {
+        var localGitInfo = new GitInfo
+        {
+            SourceRoot = gitDirectory.Parent?.FullName
+        };
+
+        gitInfo = localGitInfo;
+
+        try
+        {
+            // Ensure we have permissions to read the git directory
+            var safeDirectory = ProcessHelpers.RunCommand(
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: $"config --global --add safe.directory {gitDirectory.FullName}",
+                    workingDirectory: gitDirectory.FullName,
+                    useWhereIsIfFileNotFound: true));
+            if (safeDirectory?.ExitCode != 0)
+            {
+                localGitInfo.Errors.Add($"Error setting safe.directory: {safeDirectory?.Error}");
+            }
+
+            // Get the repository URL
+            var repositoryOutput = ProcessHelpers.RunCommand(
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: "ls-remote --get-url",
```

Comment:
What if there are multiple remotes? 🤔 Also, this will return different formats for https vs ssh e.g. `https://github.com/DataDog/dd-trace-dotnet.git` vs `git@github.com:DataDog/dd-trace-dotnet.git`,  - is that ok here?

FWIW, I tested locally, and this _doesn't_ seem to include "local" remotes, so that's good

Reply:
by default it will try to resolve `origin`. for the format it's ok, the backend is prepared to resolve both and create a `repositoryId` from that